Reuse frame state arrays

fredj opened the pull request:
Reuse the extent and viewHints arrays from the previous frameState to reduce the number of temporary array

tschaub commented on src/ol/view.js:
```diff
@@ -256,10 +256,17 @@ ol.View.prototype.getCenter = function() {
  */
-ol.View.prototype.getHints = function() {
-  return this.hints_.slice();
+ol.View.prototype.getHints = function(opt_hints) {
```
I'm not opposed to this.  But do you see any problem with just returning `this.hints_`?

fredj replied:
Not sure about this; `frameState.viewHints` is updated in (at least) the [animation functions](https://github.com/openlayers/ol3/blob/v3.15.1/src/ol/animation.js#L32)

tschaub replied:
Ok, I agree we can't reuse the same hints array everywhere.  Though I think it is a pity that view hints are specific to a frame instead of a view.

If I ever get around to changing how animation works (so view properties are properly animated and we could finally have a "side-by-side" example that look nice), then I think we won't need the extra code here.

fredj replied:
Thanks for the reminder, #5234

tschaub commented:
This looks like a good idea to me.  I didn't check where else `view.getHints()` is called, but I wonder if we could just pass around a single array instead of adding the `opt_hints` arg.

I'm also curious to learn if you are profiling to help identify places where garbage collection can be reduced.  Or if you are just identifying optimization opportunities like this by looking through the code.

fredj commented:
> I'm also curious to learn if you are profiling to help identify places where garbage collection can be reduced. Or if you are just identifying optimization opportunities like this by looking through the code.

It was very manual; I've added breakpoints in `size.js`, `extent.js`, ... where the optional value is not provided. For example with :
```js
ol.size.buffer = function(size, buffer, opt_size) {
 ...
};
```
I've added a conditional breakpoint if `opt_size == undefined`.

tschaub commented:
This looks good to merge.

fredj commented:
Thanks for the review
